cfgtool

```diff
@@ -192,56 +192,106 @@ autoinstall() {
   if test -f "${key_easyrsa_file}" && test -f "${crt_easyrsa_file}"; then
     er "${key_easyrsa_file} and ${crt_easyrsa_file} already exist. Reusing those keys. Delete/backup them if you want to use new keys"
   else
-    easyrsa build-client-full "$name" nopass
+    easyrsa build-client-full "${name}" nopass
   fi
 
+  #Check if the ca and key are generated
   test -f "${key_easyrsa_file}" && test -f "${crt_easyrsa_file}" ||
       fail "${key_easyrsa_file} and ${crt_easyrsa_file} missing"
 
   (
-    mkdir -p "$client_vpn_dirname"
-
-    expand_vars < "$VPN_DIR"/openvpn-client.cfg.proto > "$client_vpn_dirname"/openvpn-client.cfg
-    expand_vars < "$VPN_DIR"/openvpn-common.cfg.proto > "$client_vpn_dirname"/openvpn-common.cfg
-    cp "$key_easyrsa_file" "$client_vpn_dirname"/client.key
-    cp "$crt_easyrsa_file" "$client_vpn_dirname/client.crt"
-    cp "$pki_dir/ca.crt" "$client_vpn_dirname"
-
-    cd "`dirname "$client_vpn_dirname"`"
-    tar -cvf "$client_vpn_tarname" "`basename "$client_vpn_dirname"`"
+    #Create the client config directory
+    mkdir -p "${client_vpn_dirname}"
+    #Place the files in the temp directory and the client config directory
+    expand_vars < "${VPN_DIR}/openvpn-client.cfg.proto" > "${client_vpn_dirname}/openvpn-client.cfg"
+    expand_vars < "${VPN_DIR}/openvpn-common.cfg.proto" > "${client_vpn_dirname}/openvpn-common.cfg"
+    cp "${key_easyrsa_file}" "${client_vpn_dirname}/client.key"
+    cp "${crt_easyrsa_file}" "${client_vpn_dirname}/client.crt"
+    cp "${pki_dir}/ca.crt" "${client_vpn_dirname}"
   )
-
+  echo "a ${client_vpn_dirname}" "${client_vpn_dirname}/${client_vpn_zipname}" -p"${password}"
+  cd "${tmp_dir}"
+  7z a "${client_vpn_zipname}" "${client_vpn_dirname}" -p"${password}"
+  cd "${working_dir}"
   er "
-The user's configuration has been created under ${client_vpn_tarname} .
+The user's configuration has been created under ${client_vpn_zipname}.
 
-Please transmit the tar over a safe channel to it's user.
+And encrypted with the generated password.
+
+Please transmit the zip over a safe channel to it's user.
 
 In order to access the VPN the user must unpack the tar, make sure the 'nobody' group and user exists on their system and execute 'openvpn openvpn-client.cfg' inside the unpacked directory as root.
 
 Stay safe. Use crypto. Don´t drink and factorize."
 
+
+}
+
+
+#http://blog.pragbits.com/it/2015/02/09/slack-notifications-via-curl/
+__notify_slack() {
+    local SLACK_TITLE="$1"
+    local SLACK_BODY="$2"
+    local payload=$(python -c "import sys;import json;print(json.dumps({'attachments':[{'text': sys.argv[1], 'textfallback': sys.argv[1], 'color':'#00ff00', 'fields': [{'title': sys.argv[2], 'short':False}]}]}))" "${SLACK_TITLE}" "${SLACK_BODY}")
+    local SLACK_URL="https://hooks.slack.com/services/${SLACK_API}"
+    curl -X POST --data "payload=${payload}" ${SLACK_URL}
+ }
+
+#Load a textfile filled with the emails of the new users
+add_clients() {
```

Comment:
How about `add_clients_and_mail_keys`.
It is important that the caller is very clear. THIS WILL SEND A SECRET KEY TO THE USER.

Can we also provide a wrapper with a predefined key location?